app/apphandlers/setup.go

```diff
@@ -107,6 +108,15 @@ func rudderCoreBaseSetup() {
 
 	processor.RegisterAdminHandlers(&readonlyProcErrorDB)
 	router.RegisterAdminHandlers(&readonlyRouterDB, &readonlyBatchRouterDB)
+
+	go func() {
+		backendconfig.DefaultBackendConfig.WaitForConfig(context.Background())
```

Comment:
Waiting for config to send is a workaround since we don't have the workspaceID, until config is fetched. 

There is a plan to remove this need in the `v2` API.